### PR TITLE
For #26144 new address and credit cards autofill UI smoke tests

### LINKED ISSUE #26144: [UI test automation] Improve coverage of the UI smoke test suite
### Summary

Improve coverage of the [UI smoke test suite](https://testrail.stage.mozaws.net/index.php?/suites/view/3192&group_by=cases:section_id&group_order=asc&group_id=45740):
• Create new UI tests
• Add missing steps

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21090)

### PR DESCRIPTION
**Pull request by AndiAJ**
For #26144 :
New address autofill UI smoke tests, successfully ran 100x on Firebase. ✅ 
New address credit cards UI smoke tests, successfully ran 100x on Firebase. ✅ 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [ ] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #26144

### CODE REVIEW

File: app/src/androidTest/java/org/mozilla/fenix/ui/SettingsBasicsTest.kt [hunk 1]
```diff
@@ -122,4 +125,65 @@ class SettingsBasicsTest {
...
+            fillAndSaveAddress(
+                "Mozilla",
+                "Fenix",
+                "Firefox",
+                "Harrison Street",
+                "San Francisco",
+                "94105",
+                "0123",
+                "mozilla@mozilla.com"
+            )
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/SettingsBasicsTest.kt:179]: I found https://github.com/mozilla-mobile/fenix/blob/main/app/src/test/java/org/mozilla/fenix/settings/address/interactor/DefaultAddressEditorInteractorTest.kt#L33 we can borrow some of the same inputs e.g, `foo@bar.com` and the `555` phone, default to `US` country

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuAutofillRobot.kt [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    fun fillAndSaveAddress(
+        firstName: String,
+        middleName: String,
+        lastName: String,
+        streetAddress: String,
+        city: String,
+        zipCode: String,
+        phoneNumber: String,
+        emailAddress: String
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuAutofillRobot.kt:43]: We should add the country or region dropdown too

**Review by AaronMT — CHANGES_REQUESTED** [review]
Looks great Andi. Just some minor changes above.

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuAutofillRobot.kt (hunk 2 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/SettingsSubMenuAutofillRobot.kt:53]: Done !

File: app/src/androidTest/java/org/mozilla/fenix/ui/SettingsBasicsTest.kt (hunk 1 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SettingsBasicsTest.kt:183]: Done !

**Comment by AndiAJ** [comment]
Made the requested changes @AaronMT , re-ran 50x the tests on Firebase and successfully passed.

**Comment by AndiAJ** [comment]
Managed to finish the credit card autofill smoke tests as well @AaronMT .
Both successfully passed 100x on Firebase. ✅

File: app/src/androidTest/assets/pages/creditCardForm.html [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<html>
+<head>
+    <meta name="viewport" content="width=device-width">
+    <title>Credit_Card_Form</title>
+</head>
+<body>
+
```
Comment (AaronMT) [inline app/src/androidTest/assets/pages/creditCardForm.html:7]: nit: whitespace

File: app/src/androidTest/assets/pages/creditCardForm.html [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+<form>
+    <p>Card information</p>
+    <p>Card Number: <input id="cardNumber" type="text" placeholder="1234 1234 1234 1234"></p>
+    <p>Name on card: <input id="nameOnCard"type="text" placeholder="Name on card"></p>
+</form>
+</body>
```
Comment (AaronMT) [inline app/src/androidTest/assets/pages/creditCardForm.html:13]: nit: indent `<form>`

**Review by AaronMT — CHANGES_REQUESTED** [review]
Looking good, just a couple of nits

File: app/src/androidTest/java/org/mozilla/fenix/ui/SettingsBasicsTest.kt [hunk 5]
```diff
@@ -26,6 +30,12 @@ class SettingsBasicsTest {
     /* ktlint-disable no-blank-line-before-rbrace */ // This imposes unreadable grouping.
     private lateinit var mockWebServer: MockWebServer
     private val featureSettingsHelper = FeatureSettingsHelper()
+    private val creditCardNumber = "2720994326581252"
+    private val lastDigitsCreditCardNumber = "1252"
+    private val nameOnCard = "mastercard"
+    private val expirationMonth = "February"
```
[the remark this answers is not in the document]
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/SettingsBasicsTest.kt]: Done !

File: app/src/androidTest/assets/pages/addressForm.html [hunk 6]
```diff
@@ -0,0 +1,17 @@
+<html>
+<head>
+    <meta name="viewport" content="width=device-width">
+    <title>Address_Form</title>
+</head>
+<body>
+
```
[the remark this answers is not in the document]
Comment (AndiAJ) [reply app/src/androidTest/assets/pages/addressForm.html]: Done !

File: app/src/androidTest/assets/pages/addressForm.html [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+<form>
+    <p>Street Address: <input id="streetAddress" type="text"></p>
+    <p>City: <input id="city" type="text"></p>
+    <p>Zip Code: <input id="zipCode" type="text"></p>
+    <p>Country: <input id="country" type="text"></p>
+    <p>Telephone: <input id="telephone" type="text"></p>
+    <p>Email: <input id="email" type="text"></p>
+</form>
+</body>
```
[the remark this answers is not in the document]
Comment (AndiAJ) [reply app/src/androidTest/assets/pages/addressForm.html:15]: Done !

File: app/src/androidTest/assets/pages/creditCardForm.html (hunk 3 above)
Comment (AndiAJ) [reply app/src/androidTest/assets/pages/creditCardForm.html]: Done !

File: app/src/androidTest/assets/pages/creditCardForm.html (hunk 4 above)
Comment (AndiAJ) [reply app/src/androidTest/assets/pages/creditCardForm.html:12]: Done !

**Review by AaronMT — APPROVED** [review]
lgtm

**Pull request merged**